### PR TITLE
TypeScript migration

### PR DESCRIPTION
**Pull request by samryderE**
This is a full migration of the current project from JS to TS.

### CODE REVIEW

File: src/AnimateQuotes.tsx [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
+  setHasInitialTransitionRun,
+  height
+}: AppPropsAnimateQuotes) {
   const [isQuoteAnimationRunning, setIsQuoteAnimationRunning] = useState(false);
   const [quoteIndex, setQuoteIndex] = useState(0);
 
   const fadeAnim = useRef(new Animated.Value(0)).current;
   const sizeAnim = useRef(new Animated.Value(0.5)).current;
 
-  const topPosition = height * TOP_POSITION_PERCENT;
```
Comment (samryderE) [inline src/AnimateQuotes.tsx:14]: Committing to centered `AnimateQuotes` component no longer requires this positioning variable.

File: src/useIsShake.ts [hunk 2]
```diff
@@ -1,18 +1,20 @@
-import React, { useState, useEffect } from 'react';
+import { useState, useEffect } from 'react';
 import { Accelerometer } from 'expo-sensors';
+import { Subscription } from 'expo-sensors/build/Pedometer';
+
+type Poll = number | null;
+type IsShakeProps = { x: number };
 
 export function useIsShake() {
-  const [{ x }, setData] = useState({
```
Comment (samryderE) [inline src/useIsShake.ts:5]: `{x}` value is never read and initial value is not necessary for state value as `setData` is invoked by `Accelerometer` listener and passed an `Accelerometer Measurement` which will then be read.

File: src/useIsShake.ts [hunk 3]
```diff
@@ -1,18 +1,20 @@
-import React, { useState, useEffect } from 'react';
+import { useState, useEffect } from 'react';
 import { Accelerometer } from 'expo-sensors';
+import { Subscription } from 'expo-sensors/build/Pedometer';
+
+type Poll = number | null;
```
Comment (samryderE) [inline src/useIsShake.ts:5]: Typing Polls array items.

File: src/useIsShake.ts [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+type IsShakeProps = { x: number };
```
Comment (samryderE) [inline src/useIsShake.ts:6]: Serves as `Accelerometer` Measurement type.

File: src/useIsShake.ts [hunk 5]
```diff
@@ -27,10 +29,10 @@ export function useIsShake() {
     if (newX >= 0) polls[1] = newX;
 
     let change;
-    if (polls[0] !== null && polls[1] !== null) {
+    if (polls[0] && (polls[1] || polls[1] === 0)) {
```
Comment (samryderE) [inline src/useIsShake.ts:32]: Rewrote this logic to meet typeScript errors concerning possibility of `undefined` being passed to following subtraction operation in assignment of `change`

File: src/useIsShake.ts (hunk 3 above)
Comment (russellwmoore) [reply src/useIsShake.ts:5]: You can remove this comment. This is exactly what the code says

**Review by russellwmoore — APPROVED** [review]
Noice. 
```
function onWard(go : boolean) : boolean {
  return go ? 'and upward' : 'and downward'
}

return `onward ${onWard(true)}`
```

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{
```
Comment (russellwmoore) [inline tsconfig.json:1]: This config will evolve over time

**Pull request merged**